docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+Total modules: 3
++--------------------------------------------------------------------------------------------------+
+```
+
+1. James will simply enter the command `addm cs3235`
+
+   After the input is parsed as an **add module task** and executed, the `AddModuleCommand#execute()` will call `ModuleManager#add()` to add the module cs3235. In the `ModuleManager#add()` method, it will call `ModuleManager#contains()` to check if the module cs3235 exists in the `ArrayList` named `moduleList` , then it will check if the module code "cs3235" is a key in the `HashMap` named `modulesMap`, after all, it will instantiate an `Module` object with the module code "CS3235" and respective title "Computer Security", then add the object into the `moduleList`.
+
+2. James receive the following feedback:
+
+   ```
+   root :
+   addm cs3235
+   SUCCESS!! Module CS3235 Computer Security has been added.
+   
+   root :
+   lsm
+   Here are what you are looking for...
+   
+   +--------------------------------------------------------------------------------------------------+
+    NO |  MODULE CODE   |                                 MODULE TITLE
+   +--------------------------------------------------------------------------------------------------+
+    1  |     CS1231     |                             Discrete Structures
+    2  |     CS2100     |                            Computer Organisation
+    3  |     CS2113     |              Software Engineering & Object-Oriented Programming
+    4  |     CS3235     |                              Computer Security
+   +--------------------------------------------------------------------------------------------------+
+   Total modules: 4
+   +--------------------------------------------------------------------------------------------------+
+   ```
+
+
+
+Below is a *sequence diagram* to illustrate the above example scenario.  
+
+```
+	// To do Sequence diagram here
+```
+
+
+
+#### 1.2. Add Category and Task Features
+
+Since add category and add task feature are implemented in a similar pattern, they can be explained together.
+
+The add category/task enable the user to add category/task/tag into Category/Task/Tag List.(One to one correspondence, which means the user can only add category to Category List, task to Task List).
+
+When the user first request to execute the **add** command*(assuming the command format given is valid)* to add a category/task by providing its name, the application will parse the input after `addm` /`addc` command word as the name for category/task. From here, there are **three** possible outcomes:
+
+1. The specified upper parent directory does not exist. -- No category/task will be added.
+2. The name of the specified category/task already exist. -- No category/task will be added.
+3. The specified upper parent directory exist and the name of the specified category/task does not exist. -- The specified category/task will be added.
+
+#### Feature Implementation
```

Comment:
The implementation is quite clear, but have you considered alternative ways to implement this feature?
I think it would be better to make a comparison between your current choice and alternatives you may have considered.
So it's more clear to see pros and cons of your current choice.